### PR TITLE
Csv hotfixes 357 - ready to review / merge

### PR DESCRIPTION
**Pull request by wgilling**
This branch only addresses the two problems
 - System Created export and import handling
 - Title field properly decode any encoded characters such as "&amp;" or "&quote;"

The branch requires that the migration_plus.migration.csv.yml be imported.

To test
1. edit an item so that it will have an encoded character in the export file
2. make an edit to this file - can adjust the System Created date (using the same date format!) and the Title may be modified
3. export this using the Download CSV for that item
4. either copy that file into the /asu_migrate/data/csv.csv file and use `drush migrate:import csv --update` or by using the Migrate Source UI interface to select "Import/Update nodes from CSV" and upload the CSV that was exported in step 1

This branch could be used again if there are other CSV issues found, but it will likely need to be merged with the latest develop code again.

### CODE REVIEW

File: web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php [hunk 1]
```diff
@@ -27,9 +27,9 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
       $nonsort = $item->entity->field_nonsort->value . '';
       $main = $item->entity->field_main_title->value;
       $sub = $item->entity->field_subtitle->value . '';
-      $nm = $nonsort . ":" .
-        ($main != NULL ? $main : "[untitled]") . ":" .
-        $sub;
+      $nm = (($nonsort) ? $nonsort . ':' : '').
```
Comment (elizoller) [inline web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php]: Is this putting a `:` between the nonsort and the main part of the title?

**Comment by elizoller** [comment]
@wgilling left you question above

File: web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php (hunk 1 above)
Comment (wgilling) [reply web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php]: yes but only if there is a value for nonsort. I might need to go back and redo this, but in my round-trip, the import worked fine -- though I didn't give my test item any value for the nonsort. Did that do something weird in your testing?

Comment (elizoller) [reply web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php]: there just shouldn't be anything but a space between nonsort and the main part of the title.
ie if the nonsort is "The" and the main part of the title is "Awesome" then the combined title should be "The Awesome" not "The: Awesome"

Comment (wgilling) [reply web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php]: oh crap! That has to change then. Lemme just edit the code via git unless you have already.

Comment (elizoller) [reply web/modules/custom/asu_migrate/src/Plugin/Field/FieldFormatter/ComplexTitleCSVFormatter.php]: i have not edited it. go for it

**Pull request merged**